slam-libraries/viam-cartographer/src/io/file_handler.cc

```diff
@@ -19,6 +19,14 @@ namespace io {
 
 namespace fs = boost::filesystem;
 
+const std::string MakeFilenameWithTimestamp(std::string path_to_dir) {
+    std::time_t t = std::time(nullptr);
+    char timestamp[100];
+    std::strftime(timestamp, sizeof(timestamp), time_format.c_str(),
+                  std::gmtime(&t));
+    return path_to_dir + "/" + "map_data_" + timestamp + ".pbstream";
+}
+
```

Comment:
Will test this function in this ticket: https://viam.atlassian.net/browse/DATA-681